<?xml version="1.0"?>

<launch>
	<arg name="use_yolo" default="true"/>
	<arg name="image_topic_name" default="<YOUR_IMAGE_TOPIC_HERE>"/>

	<!-- YOLO Parameters -->
	<arg name="classnames_filename" default="<YOUR_CLASSNAMES_FILENAME_HERE>" />
	<arg name="cfg_filename" default="<YOUR_CFG_FILENAME_HERE>" />
	<arg name="weight_filename" default="<YOUR_WEIGHT_FILENAME_HERE>" />

	<param name="image_topic_name" value="$(arg image_topic_name)"/>
	<param name="object_detector_data_dir" value="$(arg data_directory)"/>

	<node name="object_detector_bridge" pkg="object_detector" type="gaze_predict_wrapper.py" output="screen" cwd="node">
		<param name="use_yolo" value="$(arg use_yolo)"/>
	</node>
	
	<node name="face_detector" pkg="object_detector" type="face_detector.py" output="screen" cwd="node"></node>

	<group if="$(arg use_yolo)">
		<include file="$(find rail_object_detector)/launch/darknet.launch">
			<arg name="publish_detections_topic" value="true"/>
			<arg name="use_scene_service" value="false"/>
			<arg name="use_image_service" value="true"/>
			<arg name="classnames_filename" value="$(arg classnames_filename)"/>
			<arg name="cfg_filename" value="$(arg cfg_filename)"/>
			<arg name="weight_filename" value="$(arg weight_filename)"/>
			<arg name="image_sub_topic_name" value="$(arg image_topic_name)"/>
		</include>
	</group>
</launch>
